### PR TITLE
Use annotation ion adduct charge for spectral lbirary building

### PR DESCRIPTION
**Pull request by robinschmid**
Charge of scan was wrong often

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java [hunk 1]
```diff
@@ -103,7 +104,10 @@ static SpectralLibraryEntry create(@Nullable MemoryMapStorage storage, final Sca
     SpectralLibraryEntry entry = create(storage,
         Objects.requireNonNullElse(match.getPrecursorMZ(), scan.getPrecursorMz()), dataPoints);
     // scan details
-    entry.putIfNotNull(DBEntryField.CHARGE, scan.getPrecursorCharge());
+    IonType adduct = match.getAdductType();
+    if (adduct != null) {
+      entry.putIfNotNull(DBEntryField.CHARGE, adduct.getCharge());
+    }
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java:110]: I think we made a utility method for this for this FeatureUtils.extractBestAbsoluteChargeState

Comment (SteffenHeu) [reply mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java:110]: ah but only works for a row. maybe use the match adduct and if its not defined fall back to the precursor charge

Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java]: done match>row>scan for charge extraction

File: mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java [hunk 2]
```diff
@@ -166,6 +168,14 @@ static SpectralLibraryEntry create(@Nullable MemoryMapStorage storage, final Sca
     return entry;
   }
 
+  static Integer getCharge(CompoundDBAnnotation match, FeatureListRow row, Scan scan) {
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java:171]: this method can take a FeatureAnnotation, then you could also expand the library generation to in house spectra of external libraries later on

Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/util/spectraldb/entry/SpectralLibraryEntry.java:171]: Great comment. Thats the goal in the long run.

**Comment by SteffenHeu** [comment]
fantastic

**Pull request merged**